bin/makisu/cmd/diff.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/spf13/cobra"
+	"github.com/uber/makisu/lib/docker/image"
+	"github.com/uber/makisu/lib/log"
+	"github.com/uber/makisu/lib/registry"
+	"github.com/uber/makisu/lib/snapshot"
+	"github.com/uber/makisu/lib/storage"
+	"github.com/uber/makisu/lib/tario"
+)
+
+type diffCmd struct {
+	*cobra.Command
+}
+
+func getDiffCmd() *diffCmd {
+	diffCmd := &diffCmd{
+		Command: &cobra.Command{
+			Use:                   "diff <image name> <image name>",
+			DisableFlagsInUseLine: true,
+			Short:                 "Compare docker images from registry",
+		},
+	}
+
+	diffCmd.Args = func(cmd *cobra.Command, args []string) error {
+		if len(args) != 2 {
+			return errors.New("Requires two image names as arguments")
+		}
+		return nil
+	}
+
+	diffCmd.Run = func(cmd *cobra.Command, args []string) {
+		if err := diffCmd.Diff(args[0], args[1]); err != nil {
+			log.Error(err)
+			os.Exit(1)
+		}
+	}
+
+	return diffCmd
+}
+
+func (cmd *diffCmd) Diff(image1, image2 string) error {
+	pullImage1, err := image.ParseNameForPull(image1)
+	if err != nil {
+		return fmt.Errorf("parse the first image %s: %s", pullImage1, err)
+	}
+
+	pullImage2, err := image.ParseNameForPull(image2)
+	if err != nil {
+		return fmt.Errorf("parse the second image %s: %s", pullImage2, err)
+	}
+
+	if err := initRegistryConfig(""); err != nil {
+		return fmt.Errorf("failed to initialize registry configuration: %s", err)
+	}
+
+	store1, err := storage.NewImageStore("/tmp/makisu-storage/image1")
+	if err != nil {
+		panic(err)
+	}
+
+	store2, err := storage.NewImageStore("/tmp/makisu-storage/image2")
+	if err != nil {
+		panic(err)
+	}
+
+	client1 := registry.New(store1, pullImage1.GetRegistry(), pullImage1.GetRepository())
```

Comment:
Seems like you actually need a for loop or another function here instead of doing the same thing twice.